src/manager/framework/state/constraints.go

```diff
@@ -0,0 +1,220 @@
+package state
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/Dataman-Cloud/swan/src/mesosproto/mesos"
+	"github.com/Dataman-Cloud/swan/src/utils"
+)
+
+//go:generate go tool yacc -o ./constraints_gen.go  ./constraints_parser.y
+
+var uniqWhat = []string{
+	"hostname",
+}
+
+var likeWhat = []string{
+	"hostname",
+	"agentid",
+}
+
+type ConstraintContextHolder struct {
```

Comment:
is ConstraintHolder is  better?  because the context and ctx is refer in particular to system context